addOns/pscanrulesBeta/src/test/java/org/zaproxy/zap/extension/pscanrulesBeta/InformationDisclosureSuspiciousCommentsUnitTest.java

```diff
@@ -0,0 +1,270 @@
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertEquals(0, alertsRaised.size());
+    }
+
+    @Test
+    public void shouldNotAlertWithoutSuspiciousCommentInJavaScriptResponse()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String body = "Some <script>text, nothing suspicious here...</script>\nLine 2\n";
+        HttpMessage msg = createHttpMessageWithRespBody(body, "text/javascript;charset=ISO-8859-1");
+
+        assertTrue(msg.getResponseHeader().isText());
+        assertTrue(msg.getResponseHeader().isJavaScript());
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertEquals(0, alertsRaised.size());
+    }
+
+    @Test
+    public void shouldAlertOnSuspiciousCommentInHtmlElements()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String body =
+                "<h1>Some text <script>Some Html Element todo DO something </script></h1>\n"
+                        + "<b>No script here</b>\n";
+        HttpMessage msg = createHttpMessageWithRespBody(body, "text/html;charset=ISO-8859-1");
+
+        assertTrue(msg.getResponseHeader().isText());
+        assertFalse(msg.getResponseHeader().isJavaScript());
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertEquals(1, alertsRaised.size());
+    }
+
+    @Test
+    public void shouldNotAlertWithoutSuspiciousCommentInElements()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String body =
+                "<h1>Some text <script>Some Html Element Fix: DO something </script></h1>\n"
+                        + "<b>No script here</b>\n";
+        HttpMessage msg = createHttpMessageWithRespBody(body, "text/html;charset=ISO-8859-1");
+
+        assertTrue(msg.getResponseHeader().isText());
+        assertFalse(msg.getResponseHeader().isJavaScript());
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertEquals(0, alertsRaised.size());
+    }
+
+    @Test
+    public void shouldAlertOnSuspiciousCommentInHTML()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String body =
+                "<h1>Some text <!--Some Html comment FixMe: DO something --></h1>\n"
+                        + "<b>No script here</b>\n";
+        HttpMessage msg = createHttpMessageWithRespBody(body, "text/html;charset=ISO-8859-1");
+
+        assertTrue(msg.getResponseHeader().isText());
+        assertFalse(msg.getResponseHeader().isJavaScript());
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then
+        assertEquals(1, alertsRaised.size());
+    }
+
+    @Test
+    public void shouldNotAlertWhenNoSuspiciousCommentInHtml()
```

Comment:
Same here.

Reply:
I've renamed all the test cases and fixed the changelog. I didn't do anything for the 2 comments my dad commented on. Let me know if there is anything else to change.